doc/proxy-metrics.md

```diff
@@ -133,3 +135,102 @@ request_total{
 
 [prom-format]: https://prometheus.io/docs/instrumenting/exposition_formats/#format-version-0.0.4
 [pod-template-hash]: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pod-template-hash-label
+
+# Transport-Level Metrics
+
+The following metrics are collected at the level of the underlying transport
+layer.
+
+### `connection_opens`
+
+A counter of the total number of transport connections which have been opened.
+
+### `connection_closes`
+
+A counter of the total number of transport connections which have been closed.
+
+### `connections`
+
+A gauge of the number of transport connections currently open.
+
+### `sent_bytes`
+
+A counter of the total number of sent bytes.
+
+### `received_bytes`
+
+A counter of the total number of recieved bytes.
+
+### `connection_duration_ms`
+
+A histogram of the duration of the lifetime of a connection, in milliseconds.
+
+### `connection_sent_bytes`
+
+A histogram of the number of bytes sent over the lifetime of a connection.
+This is updated when a connection closes.
+
+### `connection_recieved_bytes`
```

Comment:
i'm not convinced that `connection_sent_bytes` and `connection_received_bytes` are of particular importance, as they don't really make _sense_.  If I have a connection open for 2 days and then it closes, then in one random time window we're going to have an extremely large payload size...

I think we should leave these out.

Reply:
Okay, that's fair --- I based my initial design for this on the metrics that Finagle exposes. Happy to remove these, it'll certainly make the implementation much simpler.

Reply:
Do we still want the overall `sent_bytes`/`received_bytes` counters?

Reply:
Taking a step back: Connections and requests tend to be different -- most requests complete within one metric sampling, whereas this is less common for connections.

For now, I think we really get what we need out of the other bytes_sent bytes_recv counters